iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -291,7 +295,14 @@ private async Task Connect(IChannelHandlerContext context)
                     Debug.Assert(this.mqttTransportSettings.ClientCertificate != null);
                 }
 
-                string usernameString = $"{this.iotHubHostName}/{id}/?{ApiVersionParam}={ClientApiVersionHelper.ApiVersionString}&{DeviceClientTypeParam}={Uri.EscapeDataString(this.productInfo.ToString())}";
+                string serviceParams = ClientApiVersionHelper.ApiVersionQueryStringLatest;
+
+                if (_pnpOptions.ModelId != null)
```

Comment:
_pnpOptions**?**.ModelId != null ?

Reply:
Should we do string.IsNullOrWhitespace instead as ModelId is a string?

Reply:
sure, I like string.IsNullOrWhitespace